### PR TITLE
Add tests around TableConvert and server side printing

### PR DESCRIPTION
**Pull request by smarterclayton**
Also wire in more table printers

@fabianofranz

### CODE REVIEW

**Comment by fabianofranz** [comment]
@smarterclayton check @wanghaoran1988 PR to make sure you guys don't duplicate conversions: https://github.com/kubernetes/kubernetes/pull/49700

**Comment by smarterclayton** [comment]
I'm not adding printers, just consuming them from the server.

File: staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go [hunk 1]
```diff
@@ -1264,6 +1265,97 @@ func (t *Tester) testListNotFound(assignFn AssignFunc) {
...
+		for j, cell := range row.Cells {
+			// do not add to this test without discussion - may break clients
+			switch cell.(type) {
+			case float64, int64, int32, int, string, bool:
+			case []interface{}:
+			default:
+				t.Errorf("row %d, cell %d has an unrecognized type, only JSON serialization safe types are allowed: %T ", i, j, cell)
+			}
+		}
+		if len(row.Cells) != len(table.ColumnDefinitions) {
```
Comment (wanghaoran1988) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go]: duplication ?

Comment (smarterclayton) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go]: I was going to add more tests and left the if block.  I'll remove it.

**Comment by smarterclayton** [comment]
Any other comments?

File: staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go [hunk 2]
```diff
@@ -1264,6 +1265,97 @@ func (t *Tester) testListNotFound(assignFn AssignFunc) {
...
+		default:
+			t.Errorf("column %d has unexpected format: %q with type %q", j, column.Format, column.Type)
+		}
+		if column.Priority < 0 || column.Priority > 2 {
+			t.Errorf("column %d has unexpected priority", j, column.Priority)
+		}
+		if len(column.Description) == 0 {
+			t.Errorf("column %d has no description", j)
+		}
+		if column.Name == "Created At" && column.Type != "date" && column.Format != "" {
```
Comment (fabianofranz) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go]: Maybe

```
if column.Name == "Created At" && (column.Type != "date" || column.Format != "")
```

Comment (smarterclayton) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go]: Hrm - is that any easier to read?

**Comment by justinsb** [comment]
These tests started failing a few hours ago with Kops, on unrelated changes. Haven't yet had a chance to track down why.

**Pull request merged**